src/app/pages-block/services/courses.service.ts

```diff
@@ -15,23 +15,23 @@ export class CoursesService {
     new CoursePage(3322, 'Second', new Date(2022, 5, 17), 59, 'Not good course.'),
   ];
 
-  getCoursesList(): Array<CoursePage> {
+  getCoursesList(): CoursePage[] {
     return this.courses
   }
 
-  addCourses(course: CoursePage) {
-    this.courses.push(course)
+  addCourses(course: CoursePage): void {
+    this.courses = this.courses.concat(course)
   }
 
-  deleteCourse(id: number) {
+  deleteCourse(id: number): CoursePage[] {
     return this.courses = this.courses.filter(el => el.id !== id);
   }
 
-  getCourseById(id: number) {
+  getCourseById(id: number): CoursePage[] {
     return this.courses = this.courses.filter(course => course.id === id)
   }
 
-  updateCourse(course: CoursePage) {
-    course.topRated = !course.topRated;
+  updateCourse(course: CoursePage): void {
+    this.courses = this.courses.filter(el => el.id !== course.id).concat(course)
```

Comment:
in this case you fill cut course from list and add updated at the end but you shouldn't change order